src/checker/simulation.rs

```diff
@@ -0,0 +1,429 @@
+//! Private module for selective re-export.
+
+use crate::checker::{Checker, Expectation, Path};
+use crate::{fingerprint, CheckerBuilder, CheckerVisitor, Fingerprint, Model, Property};
+use dashmap::DashMap;
+use rand::rngs::StdRng;
+use rand::Rng;
+use rand::SeedableRng;
+use std::collections::{HashMap, HashSet, VecDeque};
+use std::hash::Hash;
+use std::num::NonZeroUsize;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
+use std::thread::JoinHandle;
+
+use super::EventuallyBits;
+
+/// Choose transitions in the model.
+///
+/// Created once for each thread.
+pub trait Chooser<M: Model>: Send + Clone + 'static {
+    /// State of the chooser during a run.
+    type State;
```

Comment:
Not a blocker: did you consider making Chooser mutable to avoid having to pass around the state? I realize it might be motivated by something that I'm simply not thinking about.

Reply:
I did consider it. It seemed to confuse configuration of the chooser with the state though, we'd have to clone the configuration for every simulation run and then clone it / synchronise it between threads.